lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -233,7 +235,13 @@ public IndexSearcher(IndexReaderContext context, Executor executor) {
             ? leaves ->
                 leaves.isEmpty()
                     ? new LeafSlice[0]
-                    : new LeafSlice[] {new LeafSlice(new ArrayList<>(leaves))}
+                    : new LeafSlice[] {
+                      new LeafSlice(
+                          new ArrayList<>(
```

Comment:
We probably could use the immutable list without allocation `new ArrayList`?

```suggestion
                      leaves.stream()
                                  .map(LeafReaderContextPartition::createForEntireSegment)
                                  .toList())
```

Reply:
sadly that is not the case because we call sort on this array and that will fail against an immutable list.

Reply:
Ah, thanks, I got confused by [1] where `Collections.singletonList` is used ... (I think non issue in this case, nothing to sort)

[1] https://github.com/apache/lucene/pull/13542/files#diff-a05ffca5b7693ec880539a5eee7bbfa2d15105e5e8a6105c064d6b66f9226318R286